rsocket-core/src/main/java/io/rsocket/lease/Lease.java

```diff
@@ -18,93 +18,91 @@
 
 import io.netty.buffer.ByteBuf;
 import io.netty.buffer.Unpooled;
-import io.rsocket.Availability;
+import java.time.Duration;
 import reactor.util.annotation.Nullable;
 
 /** A contract for RSocket lease, which is sent by a request acceptor and is time bound. */
-public interface Lease extends Availability {
+public final class Lease {
 
-  static Lease create(int timeToLiveMillis, int numberOfRequests, @Nullable ByteBuf metadata) {
-    return LeaseImpl.create(timeToLiveMillis, numberOfRequests, metadata);
+  public static Lease create(
+      Duration timeToLive, int numberOfRequests, @Nullable ByteBuf metadata) {
+    return new Lease(timeToLive, numberOfRequests, metadata);
   }
 
-  static Lease create(int timeToLiveMillis, int numberOfRequests) {
-    return create(timeToLiveMillis, numberOfRequests, Unpooled.EMPTY_BUFFER);
+  public static Lease create(Duration timeToLive, int numberOfRequests) {
+    return create(timeToLive, numberOfRequests, Unpooled.EMPTY_BUFFER);
   }
 
-  /**
-   * Number of requests allowed by this lease.
-   *
-   * @return The number of requests allowed by this lease.
-   */
-  int getAllowedRequests();
+  public static Lease unbounded() {
+    return unbounded(null);
+  }
 
-  /**
-   * Initial number of requests allowed by this lease.
-   *
-   * @return initial number of requests allowed by this lease.
-   */
-  default int getStartingAllowedRequests() {
-    throw new UnsupportedOperationException("Not implemented");
+  public static Lease unbounded(@Nullable ByteBuf metadata) {
+    return create(Duration.ofMillis(Integer.MAX_VALUE), Integer.MAX_VALUE, metadata);
+  }
+
+  public static Lease empty() {
+    return create(Duration.ZERO, 0);
+  }
+
+  final int timeToLiveMillis;
+  final int allowedRequests;
+  final ByteBuf metadata;
+  final long expireAt;
+
+  Lease(Duration timeToLive, int allowedRequests, @Nullable ByteBuf metadata) {
+    this.allowedRequests = allowedRequests;
+    this.timeToLiveMillis = (int) Math.min(timeToLive.toMillis(), Integer.MAX_VALUE);
+    this.metadata = metadata == null ? Unpooled.EMPTY_BUFFER : metadata;
+    this.expireAt = timeToLive.isZero() ? 0 : System.currentTimeMillis() + timeToLive.toMillis();
   }
 
   /**
-   * Number of milliseconds that this lease is valid from the time it is received.
+   * Number of requests allowed by this lease.
    *
-   * @return Number of milliseconds that this lease is valid from the time it is received.
+   * @return The number of requests allowed by this lease.
    */
-  int getTimeToLiveMillis();
+  public int allowedRequests() {
+    return allowedRequests;
+  }
```

Comment:
While we're changing these, let's calls this `numberOfRequests` to match the spec (and on `create` above).